pkg/asset/manifests/clusterapi/cluster.go

```diff
@@ -95,6 +96,7 @@ func (c *Cluster) Generate(dependencies asset.Parents) error {
 		},
 		Spec: clusterv1.ClusterSpec{},
 	}
+	cluster.SetGroupVersionKind(clusterv1.GroupVersion.WithKind("Cluster"))
```

Comment:
We shouldn't generally be needing to set these manually, the `Scheme` should have all these types registered, and when marshalling them, the GVK is automatically filled. If that's not the case, we should look into it a bit more